pypeit/edgetrace.py

```diff
@@ -1484,9 +1487,11 @@ def show(self, include_error=False, thin=1, in_ginga=False, include_img=False,
 
             # spat_id
             if synced:
-                half = fit.shape[0] // 2
-                scen = (_trc[half, gpm & is_left] + _trc[half, gpm & is_right]) / 2.
-                id_kwargs['slit_ids'] = scen.astype(int)
+                if slits is None:
+                    half = fit.shape[0] // 2
+                    slit_ids = ((_trc[half, gpm & is_left] + _trc[half, gpm & is_right]) / 2.).astype(int)
+
+                id_kwargs['slit_ids'] = slit_ids
```

Comment:
I changed how this is done.  The main change is that we should come out of the top if.else block with slit_ids defined for both cases instead of adding another if statement here.  Check my commit and make sure you agree.